### PR TITLE
env support in terminal api 

### LINKED ISSUE #20446: Allow terminal environment to be set when created via extension API

### PR DESCRIPTION
**Pull request by ramya-rao-a**
Fixes #20446

### CODE REVIEW

File: src/vs/vscode.d.ts [hunk 1]
```diff
@@ -4452,6 +4452,11 @@ declare module 'vscode' {
 		 * Args for the custom shell executable, this does not work on Windows (see #8429)
 		 */
 		shellArgs?: string[];
+		/**
+		 * A custom environment for the terminal, if this is not set the environment will be inherited
```
Comment (Tyriar) [inline src/vs/vscode.d.ts]: This should be clear about adding to the environment, not replacing it.

**Review by Tyriar — CHANGES_REQUESTED** [review]
Looks good, did you use a extension to verify?

File: src/vs/vscode.d.ts (hunk 1 above)
Comment (ramya-rao-a) [inline src/vs/vscode.d.ts]: We are indeed replacing the env. The `env` property on `IShellLaunchConfig` does eventually replace and not add to the VS Code process.

I can change the api to make it additive if you are ok with changing the use of `IShellLaunchConfig` as well

Comment (Tyriar) [inline src/vs/vscode.d.ts]: Asked people's thoughts in https://github.com/Microsoft/vscode/issues/20446#issuecomment-314267087

**Comment by Tyriar** [comment]
Works like a charm, thanks a lot @ramya-rao-a. Sorry about the delay on my end 😃

**Comment by chrmarti** [comment]
Thanks everyone! Will be using it with the Cloud Shell integration.

**Pull request merged**